app/controllers/preferences_controller.rb

```diff
@@ -5,4 +5,24 @@ def index
     @preferences = current_user.preferences
     @pagy, @records = pagy(@preferences)
   end
+
+  def new
+    @preference = Preference.new
+  end
+
+  def create
+    @preference = Preference.new(preference_params)
+    @preference.user = current_user
+    if @preference.save!
+      redirect_to preferences_path
```

Comment:
Como una mejora, podrias agregarle un mensaje con `notice` luego del redirect. 
En el archivo `en.yml` se definen las traducciones, que despues pueden ser usadas desde toda la app. Eso esta bueno ya que si en algun momento se requiere modificar un texto, se actualiza en un solo lugar 😊 . 
Es un poco extensa la docu sobre este tema, pero te dejo link: https://guides.rubyonrails.org/i18n.html#how-i18n-in-ruby-on-rails-works.